src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -527,14 +839,22 @@ public class Derived : Base
 {
     public override object M() => null;
 }
+public class Program
+{
+    void M(Base b, Derived d)
+    {
+        string s1 = b.M();
+        object s2 = d.M();
+    }
+}
 ";
-            var comp = CreateCompilation(source, parseOptions: TestOptions.WithoutCovariantReturns).VerifyDiagnostics(
+            var comp = CreateCompilationWithoutCovariantReturns(source).VerifyDiagnostics(
                 // (8,28): error CS0508: 'Derived.M()': return type must be 'string' to match overridden member 'Base.M()'
```

Comment:
should we change this diagnostic message to something like: "return type must be 'string' or a subtype to match.."?

Reply:
If the runtime supports it, yes.  This is already recorded as a PROTOTYPE comment at line 1023 of `SourceMemberContainerSymbol_ImplementationChecks.cs`

---
In reply to: [421058972](https://github.com/dotnet/roslyn/pull/43797#discussion_r421058972) [](ancestors = 421058972)